docs/Quiz Feature Planning.md

```diff
@@ -0,0 +1,311 @@
+          "transcript": "<transcript>"
+          "metadata": <metadata>
+        }
+        ```
+        
+    - **Response Payload:**
+        
+        ```
+        {
+          "quiz_id": "<quiz_id>",
+          "video_id": "<video_id>",
+          "questions": [
+            {
+                "question_text": "Which the options below best describes js?",
+                "options":  ["Option 1", "Option 2", "Option 3", "Option 4"],
+                "correct_answer": 1,
+                "explanation":  ["Explanation 1", "Explanation 2", "Explanation 3", "Explanation 4"],
+            },
+            {
+                "question_text": "Which the options below best describes CORS?",
+                "options":  ["Option 1", "Option 2", "Option 3", "Option 4"],
+                "correct_answer": 3,
+                "explanation":  ["Explanation 1", "Explanation 2", "Explanation 3", "Explanation 4"],
+            }, ...
+          ]
+        }
+        ```
+
+- **Endpoint:** `GET /quiz/{quiz_id}`
+
+    - **Description:** Retrieves a previously generated quiz by its unique ID.  
+
+      
+
+- **Endpoint:** `GET /quiz/`
+    - **Description:** Lists all stored quizzes.  
+
+
+ - **Endpoint:** `DELETE /quiz/{quiz_id}`
+    - **Description:** Deletes a quiz by its ID.
+
+2. **Dynamic quiz**
+
+ - **Endpoint:** `POST /generate-question`
+        
+    - **Description:** Generates a single question dynamically based on the current timestamp.
+
+    - **Request Payload:**
+        
+        ```
+        {
+          "video_id": "<video_id>",
+          "current_timestamp": <timestamp>,
+          "transcript": "<transcript>"
+          "metadata": <metadata>
+        }
+        ```
+        
+    - **Response Payload:**
+        
+        ```
+        {
+          "quiz_id": "<quiz_id>",
+          "video_id": "<video_id>",
+          "question": {
+                "question_text": "Which the options below best describes js?",
+                "options":  ["Option 1", "Option 2", "Option 3", "Option 4"],
+                "correct_answer": 1
+            }
+        }
+        ```
+
+- **Endpoint:** `GET /quiz/{quiz_id}`
+
+    - **Description:** Get explanation for the right answer and also why each wrong answer is not right.  
+
+    - **Request Payload:**
+        
+        ```
+        {
+          "question_id": "<question_id>",
+          "quiz_context": {
+                "question_text": "What is the purpose of the internet?",
+            "options": [
+                "Communication",
+                "Gaming",
+                "Education",
+                "All of the above"
+            ],
+            "correct_index": 3,
+            }
+        }
+        ```
+
+        - **Response Payload:**
+        ```
+        {
+            "explanation1": <exp1>,
+            "explanation2": <exp2>,
+            "explanation3": <exp3>,
+            "explanation4": <exp4>, 
+        }
+        ```
+
+
+
+3. **AI Service-check answer**
```

Comment:
We prolly wont have this endpoint in the ai service or at all. Checking the answer would happen in the frontend extension most likely since were storing the questions and answers in local storage.